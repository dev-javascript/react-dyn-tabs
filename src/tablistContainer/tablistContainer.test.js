import React from 'react';
import {unmountComponentAtNode} from 'react-dom';
import TablistContainer from './tablistContainer';
import renderer from 'react-test-renderer';
import Api from '../utils/api/api.js';
let container = document.createElement('div'),
  realUseContext;
const setMockUseContext = (op = {}) => {
  const instance = new Api({options: Object.assign({}, op)});
  React.useContext = jest.fn(() => instance);
};
beforeAll(() => {
  document.body.appendChild(container);
  realUseContext = React.useContext;
});
beforeEach(() => {});
afterEach(() => {
  unmountComponentAtNode(container);
  container.innerHTML = '';
  React.useContext = realUseContext;
});
afterAll(() => {
  document.body.removeChild(container);
  container = null;
});
describe('TablistContainer structure : ', () => {
  test('default options', () => {
    setMockUseContext();
    const tree = renderer
      .create(
        <div>
          <TablistContainer></TablistContainer>
        </div>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('children props', () => {
    setMockUseContext();
    const tree = renderer
      .create(
        <div>
          <TablistContainer>
            <div className="rc-dyn-tabs-tablist-overflow" />
          </TablistContainer>
        </div>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
