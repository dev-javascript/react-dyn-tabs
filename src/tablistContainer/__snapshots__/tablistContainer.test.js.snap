// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TablistContainer structure :  children props 1`] = `
<div>
  <div
    className="rc-dyn-tabs-tablist-container"
  >
    <div
      className="rc-dyn-tabs-tablist-overflow"
    />
  </div>
</div>
`;

exports[`TablistContainer structure :  default options 1`] = `
<div>
  <div
    className="rc-dyn-tabs-tablist-container"
  />
</div>
`;
